src/heartbeat/heartbeat.js

```diff
@@ -9,168 +9,53 @@
-const logger = require('winston');
-const errors = require('../errors');
 const request = require('superagent');
 const configModule = require('../config/config');
 const handleHeartbeatResponse = require('./listener').handleHeartbeatResponse;
-const generatorsDir = require('../constants').generatorsDir;
 const fs = require('fs');
-const Path = require('path');
 const Promise = require('bluebird');
 Promise.promisifyAll(fs);
 const u = require('../utils/commonUtils');
-let lastHeartbeatTime;
 
 /**
  * Send a heartbeat to the Refocus server
  * @returns {Request} - the request sent to the Refocus server
- * @throws {ValidationError} - if required config fields are missing
  */
 function sendHeartbeat() {
-  debug('Entered sendHeartbeat');
+  debug('Entered heartbeat.sendHeartbeat');
   const timestamp = Date.now();
   const config = configModule.getConfig();
   debug('sendHeartbeat config.refocus', config.refocus);
-  let collectorName;
-  let baseUrl;
-  let token;
-  let path;
-  let url;
-  try {
-    collectorName = config.name;
-    baseUrl = config.refocus.url;
-    token = config.refocus.collectorToken;
-    path = `/v1/collectors/${collectorName}/heartbeat`;
-    url = baseUrl + path;
-
-    if (collectorName == null) {
-      throw new errors.ValidationError('No collectorName in config');
-    } else if (baseUrl == null) {
-      throw new errors.ValidationError('No refocusUrl in config');
-    } else if (token == null) {
-      throw new errors.ValidationError('No collectorToken in config');
-    }
-
-    const existing = configModule.getConfig().metadata;
-    const current = u.getCurrentMetadata();
-    const changed = u.getChangedMetadata(existing, current);
-    Object.assign(existing, current);
-
-    const body = {
-      logLines: [],
-      timestamp: timestamp,
-      refocus: changed,
-    };
-
-    return buildMockResponse(generatorsDir)
-    .then(res => handleHeartbeatResponse(null, res))
-    .catch(err => handleHeartbeatResponse(err, null));
-
-    //TODO: send the real request and handle the response once the api can handle it
-    //debug(`sendHeartbeat sending request. url: ${url} body: %o`, body);
-    //
-    //const req = request.post(url)
-    //.set('Authorization', token)
-    //.send(body);
-
-    //.end((res) => {
-    //  res.timestamp = timestamp;
-    //  handleHeartbeatResponse(null, res)
-    // })
-
-    //return req;
-  }
-  catch (err) {
-    throw err;
+  const collectorName = config.name;
+  const refocusUrl = config.refocus.url;
+  const accessToken = config.refocus.accessToken;
+  const proxy = config.refocus.proxy;
+  const heartbeatEndpoint = `/v1/${collectorName}/heartbeat`;
+  const urlToPost = refocusUrl + heartbeatEndpoint;
+
+  const existing = configModule.getConfig().metadata;
+  const current = u.getCurrentMetadata();
+  const changed = u.getChangedMetadata(existing, current);
+  Object.assign(existing, current);
+  const requestbody = {
+    logLines: [],
```

Comment:
This should be set for real now that we're actually sending the heartbeat to refocus.

Reply:
The initial intent was to have the logs sent with the heartbeat but later we decided to have a buffered queue post to the v1/auditEvents endpoint directly. @iamigo Can you confirm this? I will delete this L43 once I get the confirmation.

Reply:
confirmed, shriram